typed-racket-lib/typed-racket/typecheck/tc-app/tc-app-eq.rkt

```diff
@@ -63,6 +64,23 @@
      (ret -Boolean (-PS (-is-type o (-val val)) (-not-type o (-val val))))]
     [((tc-result1: (Val-able: (? ok? val))) (tc-result1: t _ o))
      (ret -Boolean (-PS (-is-type o (-val val)) (-not-type o (-val val))))]
+
+    ;; In this case, try to find there is an overlap between t1 and t2
+    [((tc-result1: t1 _ o1) (tc-result1: t2 _ o2))
+     (=> fail)
+
+     (when (ormap (lambda (a) (id=? comparator a))
+                  (list #'member #'memv #'memq))
```

Comment:
Are these supposed to be `equal?` etc?

Reply:
My intention was to not deal with `member` `memv` `memq` for lists in this case, but to handle other binary comparators.